src/SelectionDot.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback } from 'react'
+import { runOnJS, useAnimatedReaction } from 'react-native-reanimated'
+import {
+  runSpring,
+  useValue,
+  useComputedValue,
+  Circle,
+  Group,
+  Shadow,
+} from '@shopify/react-native-skia'
+import type { SelectionDotProps } from './LineGraphProps'
+
+export function SelectionDot({
+  isActive,
+  color,
+  circleX,
+  circleY,
+}: SelectionDotProps): React.ReactElement {
+  const circleRadius = useValue(0)
+  const circleStrokeRadius = useComputedValue(
+    () => circleRadius.current * 6,
+    [circleRadius]
+  )
+
+  const setIsActive = useCallback(
+    (active: boolean) => {
+      runSpring(circleRadius, active ? 5 : 0, {
+        mass: 1,
+        stiffness: 1000,
+        damping: 50,
+        velocity: 0,
+      })
```

Comment:
I'd move that into the useAnimatedReaction, no need to add an extra callback here